feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -205,14 +206,34 @@ class BillingWrapper(
         presentedOfferingIdentifier: String?
     ) {
         val googleProduct = storeProduct.googleProduct
+
         if (googleProduct == null) {
-            val errorMessage = "Product must be a Google Product."
-            errorLog(errorMessage)
-            purchasesUpdatedListener?.onPurchasesFailedToUpdate(
-                PurchasesError(PurchasesErrorCode.UnknownError, errorMessage)
+            val error = PurchasesError(
```

Comment:
this seems like a lot of validation to keep in this function, perhaps we should make the `buildPurchaseParams` have a callback and return an error we can handle here?

Reply:
Yeah ... I thought about that option. I didn't like it too much since it gave me the impression the function was performing some asynchronous task or something but I think it should be ok. One alternative would be to have a `Result` kind of class (similar to purchases-ios) with the result or an error or something like that. I think either option could work.